### PR TITLE
Osd message showtime

### PR DESCRIPTION
**Pull request by vierfuffzig**
adds parameter OSDn_MSG_TIME to set message item's show duration in seconds

### CODE REVIEW

File: libraries/AP_OSD/AP_OSD.h [hunk 1]
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
     AP_Int8 enabled;
     AP_Int16 channel_min;
     AP_Int16 channel_max;
+    AP_Int8 msgtime;
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.h]: it would be good to make the units clear in the name, so maybe msgtime_s ?

File: libraries/AP_OSD/AP_OSD.h [hunk 2]
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
...
+
+    AP_OSD_Screen(int8_t msgtime);
 
 private:
     AP_OSD_Backend *backend;
     AP_OSD *osd;
 
-    static const uint16_t message_show_time_ms = 20000;
+    static const uint16_t message_show_time_ms = 1000;
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.h]: why do we need this variable now?

File: libraries/AP_OSD/AP_OSD.h [hunk 3]
```diff
@@ -59,12 +59,15 @@ class AP_OSD_Screen {
     AP_Int8 enabled;
     AP_Int16 channel_min;
     AP_Int16 channel_max;
+    AP_Int8 msgtime;
+
+    AP_OSD_Screen(int8_t msgtime);
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD.h]: what is this change for?

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 4]
```diff
@@ -153,6 +153,13 @@ const AP_Param::GroupInfo AP_OSD_Screen::var_info[] = {
     // @Path: AP_OSD_Setting.cpp
     AP_SUBGROUPINFO(pitch_angle, "PITCH", 27, AP_OSD_Screen, AP_OSD_Setting),
 
+    // @Param: MSG_TIME
+    // @DisplayName: Message display duration in seconds
+    // @Description: Sets message duration seconds
+    // @Range: 1 20
+    // @User: Standard
+    AP_GROUPINFO("MSG_TIME", 28, AP_OSD_Screen, msgtime, 10),
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp]: this is adding it per-screen. Is that needed?

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 5]
```diff
@@ -422,9 +429,10 @@ void AP_OSD_Screen::draw_batused(uint8_t x, uint8_t y)
 void AP_OSD_Screen::draw_message(uint8_t x, uint8_t y)
 {
     AP_Notify * notify = AP_Notify::instance();
+    uint16_t duration = message_show_time_ms * msgtime;
```
Comment (tridge) [inline libraries/AP_OSD/AP_OSD_Screen.cpp]: I think you're just conventing from s to ms here? Just multiple by constant 1000 instead of using a variable

**Comment by vierfuffzig** [comment]
thanks @tridge for helping a noob to figure things! 
moved from per-screen to general
deleted useless defs and variables
renamed variable respectively.

**Comment by sh83** [comment]
Looks good, thank you.

**Comment by tridge** [comment]
I've squashed the commits and rebased

**Pull request merged**